pkg/utils/kubernetes/utils.go

```diff
@@ -0,0 +1,33 @@
+package kubernetes
+
+import (
+	"errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+func CustomResourceDefinitionExists(gvk schema.GroupVersionKind) error {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return err
+	}
+	client, err := discovery.NewDiscoveryClientForConfig(cfg)
+	if err != nil {
+		return err
+	}
+	return customResourceDefinitionExists(gvk, client)
+}
+
+func customResourceDefinitionExists(gvk schema.GroupVersionKind, client discovery.ServerResourcesInterface) error {
+	api, err := client.ServerResourcesForGroupVersion(gvk.GroupVersion().String())
+	if err != nil {
+		return err
+	}
+	for _, a := range api.APIResources {
+		if a.Kind == gvk.Kind {
+			return nil
```

Comment:
IIRC, we're disregarding the Group & Version and saying that a CRD exists based solely on Kind, which may not be unique if multiple versions or installations exist. Is this problematic? Should we be checking that all three GVK match instead of just K?

Reply:
oh, in this code? No, we're only fetching resources for the group and version of interest, so by the time we get to line 27, we know that GV already matches.

Reply:
Ahh right, I'm with you. I see it.